### ISSUE TITLE
Vulnerability Remediations - Hotfix Mitigation

### ISSUE THREAD

**Issue by tschmidtb51** [oasis-tcs/csaf#187]
This issue is meant to provide more details to the **Vulnerability Remediations - Hotfix Mitigation** proposed in #185 by @wrideout.

---

## Vulnerability Remediations

**Hotfix Mitigation**

Currently, the CSAF2.0 specification does not afford users with a clear section for describing hotfix information, outside of specifying a mitigation with type "mitigation".  The creation of a new hotfix type would allow fo the storage and use of this information.  The hotfix type would include the following member attributes:

- Hotfix Name: text containing the hotfix package name.
- Hotfix Version: text containing the hotfix package version number.
- Versions Fixed: a comma-separated list specifying the software versions to which the hotfix is applicable.
- Hotfix Hash: an optional node containing the hash of the hotfix file, for hotfix package verification purposes.  This node contains the following attributes:
   - Hash Type: an enum which specifies the type of the hash, i.e. SHA-256, SHA-384, SHA-512, etc.
   - Hash Value: text of the actual hash of the hotfix file.

The type as described above would be an optional attribute of the remediations item type.  If this optional attribute is used, the type of the enclosing remediation item should be a new enum value "hotfix".  An example of this new type is as follows:

```
"vulnerabilities": [
    {
      "references": [
        {
          "type": "self", 
          "url": "https://www.example.com/", 
          "summary": "CVE-0000-00000"
        }
      ], 
      "remediations": [ 
        {
          "type": "hotfix", 
          "details": "description of the hotfix",
          "hotfix": {
             "name": "Some hotfix name",
             "version": "1.0.0",
             "vers_fixed": [
                "XXX",
                "YYY"
             ],
             "hash" : {
                "type": "SHA-512",
                "value": "1234567890deadbeef"
             }
          }
        }
      ],
...
```

**Comment by tschmidtb51** [comment]
Thank you for your contribution.

IMHO: We should not introduce optional attributes which become required if a certain other type is selected since JSON does not support inheritance. Having the attribute only optional could lead to situations where we have the `hotfix` property present inside a `none_available` type. This is valid from the point of the JSON schema but does not make sense logically. However, this could be solved by additional constraints in the standard (which everyone has to implement...). Moreover, I think the suggestion introduces here information which doesn't belong here since we already have places for this.

Therefore, I suggest to use the fields which we already have. Please see the example (excerpt) below:
```
  "product_tree": {
    "branches": [
      {
        "name": "Example Company",
        "type": "vendor",
        "branches": [
          {
            "name": "Family A",
            "type": "product_family",
            "branches": [
              {
                "name": "Product B",
                "type": "product_name",
                "branches": [
                  {
                    "name": "C.D",
                    "type": "product_version",
                    "product": {
                      "product_id": "CSAFPID-0001",
                      "name": "Example Company Family A Product B C.D"
                    }
                  },
                  {
                    "name": "C.F",
                    "type": "product_version",
                    "product": {
                      "product_id": "CSAFPID-0003",
                      "name": "Example Company Family A Product B C.F"
                    }
                  },
                  {
                    "name": "C.G",
                    "type": "product_version",
                    "product": {
                      "product_id": "CSAFPID-0004",
                      "name": "Example Company Family A Product B C.G"
                    }
                  }
                ]
              }
            ]
          }
        ]
      }
    ],
    "relationships": [
      {
        "full_product_name": {
          "name": "Example Company Family A Product B C.D installed with Hotfix E",
          "product_id": "CSAFPID-0002"
        },
        "product_reference": "CSAFPID-0001",
        "relates_to_product_reference": "CSAFPID-0005",
        "relationship_type": "installed_with"
      }
    ],
    "full_product_names": [
      {
        "product_id": "CSAFPID-0005",
        "name": "Hotfix E for Example Company Family A Product B C.D"
      }
    ]
  },
  "vulnerabilities": [
    {
      "product_status": {
        "known_affected": [
          "CSAFPID-0001"
        ],
        "first_fixed": [
          "CSAFPID-0002"
        ],
        "fixed": [
          "CSAFPID-0003"
        ],
        "recommended": [
          "CSAFPID-0004"
        ]
      },
      "remediations": [
        {
          "details": "Hotfix E: more details...",
          "type": "mitigation",
          "product_ids": [
            "CSAFPID-0001"
          ],
          "url": "https://example.com/retrieve-hotfix/hotfix-e",
          "date": "2021-02-11T11:00:00.000Z"
        },
        {
          "details": "Point here to your fixed version.",
          "type": "vendor_fix",
          "product_ids": [
            "CSAFPID-0001"
          ],
          "url": "https://example.com/download/update/C.F",
          "date": "2021-02-18T11:00:00.000Z"
        },
        {
          "details": "Point here to your recommended update.",
          "type": "vendor_fix",
          "product_ids": [
            "CSAFPID-0001"
          ],
          "url": "https://example.com/download/update/C.G",
          "date": "2021-02-25T11:00:00.000Z"
        }
      ]
    }
  ]
}

**Comment by tschmidtb51** [comment]
The Hotfix gets listed as a product (`CSAFPID-0005`), once it is install with the vulnerable product (`CSAFPID-0001`) it become a new product (`CSAFPID-0002`) - the one with the Hotfix installed. You're then able to communicate over the mitigation type as well as the `product_status` which version a customer can / should use.

**Comment by tschmidtb51** [comment]
#126 (#184) provides the opportunity to also give hashes. 
If the Hotfix applies to more than one product you could simply add an additional relationship item.

**Comment by wrideout** [comment]
Thanks for clarifying how we could leverage the existing structure-- I think that's an acceptable resolution to what I was suggesting, once hashes can be included.

**Comment by tschmidtb51** [comment]
@wrideout: hashes have been included with #197 / #126.

**Comment by sthagen** [comment]
So, I suggest we close this issue as resolved. Or did I misinterpret the discussion?
